src/wi_stuff.c

```diff
@@ -2218,8 +2270,31 @@ void WI_Start(wbstartstruct_t* wbstartstruct)
 {
   WI_initVariables(wbstartstruct);
 
-  exitpic = (wbs->lastmapinfo && wbs->lastmapinfo->exitpic[0]) ? wbs->lastmapinfo->exitpic : NULL;
-  enterpic = (wbs->nextmapinfo && wbs->nextmapinfo->enterpic[0]) ? wbs->nextmapinfo->enterpic : NULL;
+  exitpic = NULL;
+  enterpic = NULL;
+
+  if (wbs->lastmapinfo)
+  {
+      if (wbs->lastmapinfo->exitpic[0])
+      {
+          exitpic = wbs->lastmapinfo->exitpic;
+      }
+      if (wbs->lastmapinfo->exitanim[0])
+      {
+          WI_ParseInterlevel(wbs->lastmapinfo->exitanim, &interlevel_exiting);
+          exitpic = interlevel_exiting.background_lump;
+      }
+
+      if (wbs->nextmapinfo->enterpic[0])
+      {
+          enterpic = wbs->nextmapinfo->enterpic;
+      }
+      if (wbs->lastmapinfo->enteranim[0])
```

Comment:
Shouldn't this be `wbs->nextmapinfo->enteranim` here and two lines later?

Reply:
Thanks, fixed here: 68a8f3a

Reply:
It still checks for `wbs->lastmapinfo` here.